orangemiddleware/ipfilter.go

```diff
@@ -0,0 +1,46 @@
+package orangemiddleware
+
+import (
+	"net"
+	"net/http"
+	"sort"
+)
+
+func IpFilter(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// go-chi's middleware/realip.go already parses for RealIp and xForwardedFor, further sets RemoteAddr to xForwardedFor ip if available.
+		ipAddress, _, splitHostPortError := net.SplitHostPort(r.RemoteAddr)
+		if splitHostPortError != nil {
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+
+		parsedIp := net.ParseIP(ipAddress)
+		if parsedIp == nil {
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+
+		if checkIfIpIsBlocked(parsedIp.String()) {
+			w.WriteHeader(http.StatusUnauthorized)
```

Comment:
write to response like this: `http.Error(w, http.StatusText(404), 404)` I dunno what the appropriate http code is, so find that out! Bad request (400) is generic, so maybe we can use that unless you find a better one!

Reply:
sure, ive added the appropriate ones